pkg/posthog/init.go

```diff
@@ -6,35 +6,36 @@ import (
 	"runtime"
 
 	"github.com/allero-io/allero/pkg/configurationManager"
+	"github.com/allero-io/allero/pkg/rulesConfig"
 	"github.com/posthog/posthog-go"
 )
 
 type PosthogClientDependencies struct {
-	ConfigurationManager *configurationManager.ConfigurationManager
-	CliVersion           string
+	RulesConfig *rulesConfig.RulesConfig
+	CliVersion  string
 }
 type PosthogClient struct {
 	cliVersion string
 	userConfig *configurationManager.UserConfig
 }
 
 func New(deps *PosthogClientDependencies) (*PosthogClient, error) {
-	userConfig, isNewUser, err := deps.ConfigurationManager.GetUserConfig()
+	userConfig, isNewUser, err := deps.RulesConfig.ConfigurationManager.GetUserConfig()
```

Comment:
accessing a sub-dependency of a package instead of directly adding the package as a dependency is bad practice. You can easily cause a dependency loop that will fail in runtime.

Instead, you can add a `GetEmail` function to `ConfigurationManager` that will do the parse token logic inside.
Also, looks like the `ParseToken` logic should sit in `ConfigurationManager`

Reply:
done